tree_hugger/core/parser/php/php_parser.py

```diff
@@ -37,17 +44,83 @@ def get_all_class_method_names(self) -> List:
                 ret_struct[current_key].append(match_from_span(tpl[0], self.splitted_code))
         return ret_struct
 
-    def get_all_function_names(self) -> List:
+    def get_all_class_names(self) -> List[str]:
         """
-        Gets all function names from a file.
-
-        It excludes all the methods, i.e. functions defined inside a class
+        Returns a list of all class names present in a file
         """
-        captures = self._run_query_and_get_captures('all_function_names', self.root_node)
-        all_funcs = set([match_from_span(n[0], self.splitted_code) for n in captures])
+        captures = self._run_query_and_get_captures('all_class_names', self.root_node)
+        return [match_from_span(t[0], self.splitted_code) for t in captures]
+        
+    def get_all_function_bodies(self) -> Dict:
+        """
+        Returns a dict where function names are the key and the whole function code are the values
 
-        methods = self.get_all_class_method_names()
-        all_methods = set([method_name  for key, value in methods.items() for method_name in value])
+        Excludes any methods, i.e., functions defined inside a class.
+        """
+        function_names = self.get_all_function_names()
+        func_and_params = self.function_names_with_params()
+        
+        captures = self._run_query_and_get_captures('all_function_bodies', self.root_node)
+        
+        ret_struct = {}
+        pp = {}
```

Comment:
I had probably wrote this variable name. Not the best one. If you wish you can change it.